DOC, TST: enable refguide_check in circleci

pdebuyl opened the pull request:
Enable CI testing of refguide_check

* Add entry in `.circleci/config.yml`
* install pandas in circleci environment
* fix traceback message to pass refguide_check in doc/source/user/quickstart.rst `python tools/refguide_check.py --rst doc/source/user/quickstart.rst` fails because of an update of the exception message. This prevents running `python tools/refguide_check.py --rst` as part of the continuous integration.

See https://github.com/numpy/numpy/issues/14970 for background.

pdebuyl committed:
DOC: fix traceback message to pass refguide_check

mattip commented:
It would be fine to add a CI run in this PR if you want, it could go in the `.circleci/config.yml` since it already has scipy and matplotlib needed for building docs. The idea would be to add another `- run:` stanza

pdebuyl committed:
DOC, TST: add refguide_check to circleci

pdebuyl commented:
I was going to ask about that in the parent issue :-)

pdebuyl committed:
install pandas for doctests in circleci

pdebuyl committed:
DOC: set pandas max_columns=10 in absolute_beginners

Without max_columns setting, pandas truncates the dataframe display in
the continuous integration with circleci.

mattip commented on .circleci/config.yml:
```diff
@@ -50,6 +51,11 @@ jobs:
+          name: run doctests on documentation
+          command: |
+            . venv/bin/activate
+            python tools/refguide_check.py --rst
```
I think this should be before the "build devdocs" step, otherwise it wastes cycles checking the documentation generated by `autodoc` from docstrings. Those are checked in a different refguide-check run on Azure, so no need to double check.

pdebuyl replied:
check

mattip reviewed (commented):
Looks good. Just one tweak needed.

pdebuyl committed:
DOC: move refguide_check before devdocs in circleci

mattip commented on .circleci/config.yml:
```diff
@@ -33,6 +33,7 @@ jobs:
             pip install cython
             pip install .
             pip install scipy
+            pip install pandas
```
it seems line 58 `git submodule update --init` should go here

pdebuyl replied:
check

pdebuyl committed:
DOC: move git submodule command up

This is needed because of the move of the refguide_check

mattip commented:
Thanks @pdebuyl for seeing this all the way through

pdebuyl commented:
Thanks for following, I am glad I could help close an issue for NumPy!
